File: neo4j/jruby/read/neo4j_read.rb
#! /usr/bin/jruby
# -*- coding: utf-8 -*-
#
#	neo4j_read.rb
#
#					Jan/31/2012
#
require 'rubygems'
require 'open-uri'
#
require 'json'
#
# ---------------------------------------------------------------------
def	row_parser (key)
	url_in = 'http://localhost:7474/db/data/node/' + key.to_s + '/properties'
	response= open(url_in,:proxy=>nil)
	str_json=response.read
#
	hhh=JSON.parse(str_json)
#
	print hhh["id"],"\t",hhh["name"],"\t",hhh["population"], \
		"\t",hhh["date_mod"]
#
	puts 
end

#
# ---------------------------------------------------------------------
puts	"*** 開始 ***"
#
for key in 2..10
	row_parser(key)
end
#json_display_proc(str_json)
#
puts	"*** 終了 ***"
#
# ---------------------------------------------------------------------

File: neo4j/rhino/read/neo4j_read.js
#! /usr/bin/java org.mozilla.javascript.tools.shell.Main
// ------------------------------------------------------------------
//
//	neo4j_read.js
//
//						Aug/30/2012
//
// ------------------------------------------------------------------
importPackage (java.io);
importPackage (org.apache.commons.httpclient);
importPackage (org.apache.commons.httpclient.methods);

load ("/var/www/data_base/common/rhino_common/net_manipulate.js");
// ------------------------------------------------------------------

print	("*** 開始 ***");

var keys = ["1","2","3","4","5","6","7","8","9"];


for (var it in keys)
	{
	var key = keys[it];
	url_in="http://localhost:7474/db/data/node/" + key + "/properties";

	lines = rest_get_proc (url_in,"text/json");
	json_parser (lines);
	}



print	("*** 終了 ***");

// ------------------------------------------------------------------
function json_parser (str_json)
{
	var unit_aa = JSON.parse (str_json);

	if ("id" in unit_aa)
		{
		var out_str = unit_aa.id + "\t" + unit_aa.name + "\t";
		out_str += unit_aa.population + "\t" + unit_aa.date_mod;
		print (out_str);
		}
}

// ------------------------------------------------------------------
